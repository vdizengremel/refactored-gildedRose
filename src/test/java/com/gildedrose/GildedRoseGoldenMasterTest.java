package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;

class GildedRoseGoldenMasterTest {

    @Test
    void shouldDecreaseSellInWhenAnyNameAndQualityIsZero() {
        OriginalItem[] items = generateItems().toArray(new OriginalItem[0]);
        GildedRoseOriginal app = new GildedRoseOriginal(items);

        for (int i = 0; i < 50; i++) {
            app.updateQuality();
        }

        Approvals.verify(Arrays.stream(items).map(OriginalItem::toString).collect(Collectors.toList()));
    }

    private List<OriginalItem> generateItems() {
        Random random = new Random(50);
        List<OriginalItem> items = new ArrayList<>();
        for (int i = 0; i < 30; i++) {
            items.addAll(createItemsOfEachType(random.nextInt(), random.nextInt()));
        }

        return items;
    }

    private List<OriginalItem> createItemsOfEachType(int sellIn, int quality) {
        return List.of(
                new OriginalItem("Foo", sellIn, quality),
                new OriginalItem("Aged Brie", sellIn, quality),
                new OriginalItem("Backstage passes to a TAFKAL80ETC concert", sellIn, quality),
                new OriginalItem("Sulfuras, Hand of Ragnaros", sellIn, quality)
        );
    }
}
