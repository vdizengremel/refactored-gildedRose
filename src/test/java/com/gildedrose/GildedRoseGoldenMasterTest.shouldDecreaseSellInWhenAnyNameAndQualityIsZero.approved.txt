[Foo, -1160871111, -1727040520, Aged Brie, -1160871111, -1727040420, Backstage passes to a TAFKAL80ETC concert, -1160871111, 0, Sulfuras, Hand of Ragnaros, -1160871061, -1727040520, Foo, -1657178959, -765924271, Aged Brie, -1657178959, -765924171, Backstage passes to a TAFKAL80ETC concert, -1657178959, 0, Sulfuras, Hand of Ragnaros, -1657178909, -765924271, Foo, -1625295844, 2102979023, Aged Brie, -1625295844, 2102979123, Backstage passes to a TAFKAL80ETC concert, -1625295844, 0, Sulfuras, Hand of Ragnaros, -1625295794, 2102979123, Foo, -2137358074, -1275367180, Aged Brie, -2137358074, -1275367080, Backstage passes to a TAFKAL80ETC concert, -2137358074, 0, Sulfuras, Hand of Ragnaros, -2137358024, -1275367180, Foo, -942771114, -900205280, Aged Brie, -942771114, -900205180, Backstage passes to a TAFKAL80ETC concert, -942771114, 0, Sulfuras, Hand of Ragnaros, -942771064, -900205280, Foo, -1057414946, -634725072, Aged Brie, -1057414946, -634724972, Backstage passes to a TAFKAL80ETC concert, -1057414946, 0, Sulfuras, Hand of Ragnaros, -1057414896, -634725072, Foo, 2080516151, 1136568261, Aged Brie, 2080516151, 1136568311, Backstage passes to a TAFKAL80ETC concert, 2080516151, 1136568311, Sulfuras, Hand of Ragnaros, 2080516201, 1136568311, Foo, 1525467207, -84358912, Aged Brie, 1525467207, -84358862, Backstage passes to a TAFKAL80ETC concert, 1525467207, -84358862, Sulfuras, Hand of Ragnaros, 1525467257, -84358912, Foo, -724320241, 1946662915, Aged Brie, -724320241, 1946663015, Backstage passes to a TAFKAL80ETC concert, -724320241, 0, Sulfuras, Hand of Ragnaros, -724320191, 1946663015, Foo, -502894915, -2012085511, Aged Brie, -502894915, -2012085411, Backstage passes to a TAFKAL80ETC concert, -502894915, 0, Sulfuras, Hand of Ragnaros, -502894865, -2012085511, Foo, -59961835, -1171183450, Aged Brie, -59961835, -1171183350, Backstage passes to a TAFKAL80ETC concert, -59961835, 0, Sulfuras, Hand of Ragnaros, -59961785, -1171183450, Foo, 814281132, 1920145993, Aged Brie, 814281132, 1920146043, Backstage passes to a TAFKAL80ETC concert, 814281132, 1920146043, Sulfuras, Hand of Ragnaros, 814281182, 1920146043, Foo, 906139275, -5308728, Aged Brie, 906139275, -5308678, Backstage passes to a TAFKAL80ETC concert, 906139275, -5308678, Sulfuras, Hand of Ragnaros, 906139325, -5308728, Foo, -907633869, 362731711, Aged Brie, -907633869, 362731811, Backstage passes to a TAFKAL80ETC concert, -907633869, 0, Sulfuras, Hand of Ragnaros, -907633819, 362731811, Foo, -2079097314, 1845599472, Aged Brie, -2079097314, 1845599572, Backstage passes to a TAFKAL80ETC concert, -2079097314, 0, Sulfuras, Hand of Ragnaros, -2079097264, 1845599572, Foo, 1070190890, -650223741, Aged Brie, 1070190890, -650223691, Backstage passes to a TAFKAL80ETC concert, 1070190890, -650223691, Sulfuras, Hand of Ragnaros, 1070190940, -650223741, Foo, -514884974, 1308659888, Aged Brie, -514884974, 1308659988, Backstage passes to a TAFKAL80ETC concert, -514884974, 0, Sulfuras, Hand of Ragnaros, -514884924, 1308659988, Foo, 290140262, -228992747, Aged Brie, 290140262, -228992697, Backstage passes to a TAFKAL80ETC concert, 290140262, -228992697, Sulfuras, Hand of Ragnaros, 290140312, -228992747, Foo, -195516002, -1968969683, Aged Brie, -195516002, -1968969583, Backstage passes to a TAFKAL80ETC concert, -195516002, 0, Sulfuras, Hand of Ragnaros, -195515952, -1968969683, Foo, -581461957, -1663829140, Aged Brie, -581461957, -1663829040, Backstage passes to a TAFKAL80ETC concert, -581461957, 0, Sulfuras, Hand of Ragnaros, -581461907, -1663829140, Foo, -644779500, 1529046538, Aged Brie, -644779500, 1529046638, Backstage passes to a TAFKAL80ETC concert, -644779500, 0, Sulfuras, Hand of Ragnaros, -644779450, 1529046638, Foo, -261297599, -910048085, Aged Brie, -261297599, -910047985, Backstage passes to a TAFKAL80ETC concert, -261297599, 0, Sulfuras, Hand of Ragnaros, -261297549, -910048085, Foo, 1585763141, 769927866, Aged Brie, 1585763141, 769927916, Backstage passes to a TAFKAL80ETC concert, 1585763141, 769927916, Sulfuras, Hand of Ragnaros, 1585763191, 769927916, Foo, -889421221, -354734789, Aged Brie, -889421221, -354734689, Backstage passes to a TAFKAL80ETC concert, -889421221, 0, Sulfuras, Hand of Ragnaros, -889421171, -354734789, Foo, -897988958, -1272800562, Aged Brie, -897988958, -1272800462, Backstage passes to a TAFKAL80ETC concert, -897988958, 0, Sulfuras, Hand of Ragnaros, -897988908, -1272800562, Foo, 355379957, 2111357062, Aged Brie, 355379957, 2111357112, Backstage passes to a TAFKAL80ETC concert, 355379957, 2111357112, Sulfuras, Hand of Ragnaros, 355380007, 2111357112, Foo, -441935962, -1630558876, Aged Brie, -441935962, -1630558776, Backstage passes to a TAFKAL80ETC concert, -441935962, 0, Sulfuras, Hand of Ragnaros, -441935912, -1630558876, Foo, -290287220, -1084403157, Aged Brie, -290287220, -1084403057, Backstage passes to a TAFKAL80ETC concert, -290287220, 0, Sulfuras, Hand of Ragnaros, -290287170, -1084403157, Foo, 1452496371, 1156953420, Aged Brie, 1452496371, 1156953470, Backstage passes to a TAFKAL80ETC concert, 1452496371, 1156953470, Sulfuras, Hand of Ragnaros, 1452496421, 1156953470, Foo, -906330253, 1800027930, Aged Brie, -906330253, 1800028030, Backstage passes to a TAFKAL80ETC concert, -906330253, 0, Sulfuras, Hand of Ragnaros, -906330203, 1800028030]
